src/main/kotlin/com/exactpro/th2/check1/entities/Checkpoint.kt

```diff
@@ -0,0 +1,12 @@
+package com.exactpro.th2.check1.entities
+
+import com.datastax.driver.core.utils.UUIDs.timeBased
+import com.exactpro.th2.check1.SessionKey
+
+data class Checkpoint(
+    val id: String = timeBased().toString(), val sessionKeyToCheckpointData: Map<SessionKey, CheckpointData>
+) {
+    constructor(sessionKeyToCheckpointData: Map<SessionKey, CheckpointData>) : this(
+        timeBased().toString(), sessionKeyToCheckpointData
+    )
```

Comment:
There is no need to have a secondary constructor. You can add @JvmOverlods annotation on the primary one. By the way, why do you need a secondary constructor like this? I did not find any place where you use it